DEMO only: reproduce node many to many msg fail

oetyng opened the pull request:
For demonstrating this issue. Not for merge (at least not now).

A client sends a `ClientPing` to close group nodes of a given name.
Those nodes send a `NodePing` to close group nodes of the hash of the original name.
A node that receives a `NodePing`, will return `NodePong(Ok(()))` if it considers itself to be among the close group nodes of the given name.

The nodes receiving a `ClientPing`, will return a `NodePong(Ok(()))` if a majority of the nodes they queried, returned a `NodePong(Ok(()))`.

The test will succeed if the client receives a majority of `NodePong(Ok(()))`.

connec commented on src/types/keys/signature.rs:
```diff
@@ -36,7 +37,12 @@ pub struct SignatureShare {
-    Ed25519(#[debug(with = "Self::fmt_ed25519")] ed25519_dalek::Signature),
+    Ed25519(
+        #[debug(with = "Self::fmt_ed25519")]
+        #[serde(deserialize_with = "Signature::deserialize_ed25519")]
```
`Self` doesn't work here, since internally serde generates a wrapper struct, which `Self` ends up referring to 😢

connec commented:
I've merged this, but there's some activity in the `ed25519` that could see this fixed or simplified upstream.

oetyng committed:
test: reproduce node many to many msg fail

oetyng committed:
chore: add node_many_to_many test to ci

oetyng committed:
test: add small delay so client finds nodes

oetyng committed:
test: incr sleep before client ops due to slow ci

oetyng committed:
refactor: remove mut access in node msg handling

- This should decrease the cases of sequential access or deadlocks.

oetyng committed:
feat: add spend storage sync mechanisms

oetyng committed:
test: add concurrent spend write read tests

oetyng committed:
fix: remove more mut access in node msg handling

- This should decrease the cases of sequential access or deadlocks.

oetyng committed:
test: improve outcome logging detail

oetyng committed:
feat: impl get closest local peers

oetyng committed:
chore: add more details to log of lt close group

oetyng committed:
refactor: use local closest in some msg comms

- We should not be required to query our peers every single time we want
the closest peers.

oetyng committed:
chore: temp incr of timeout and keep alive vals
